crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -300,4 +303,109 @@ impl BeaconState {
             signature: attestation.signature,
         })
     }
+
+    /// Increase the validator balance at index ``index`` by ``delta``.
+    pub fn increase_balance(&mut self, index: u64, delta: u64) {
+        if let Some(balance) = self.balances.get_mut(index as usize) {
+            *balance += delta;
+        }
+    }
+
+    /// Decrease the validator balance at index ``index`` by ``delta`` with underflow protection.
+    pub fn decrease_balance(&mut self, index: u64, delta: u64) {
+        if let Some(balance) = self.balances.get_mut(index as usize) {
+            let _ = balance.saturating_sub(delta);
+        }
+    }
+
+    /// Initiate if validator already initiated exit.
+    pub fn initiate_validator_exit(&mut self, index: u64) {
+        if index as usize >= self.validators.len() {
+            return;
+        }
+        let exit_epochs: Vec<u64> = self
+            .validators
+            .iter()
+            .filter_map(|v| {
+                if v.exit_epoch != FAR_FUTURE_EPOCH {
+                    Some(v.exit_epoch)
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let current_epoch = self.get_current_epoch();
+        let mut exit_queue_epoch = exit_epochs
+            .into_iter()
+            .max()
+            .unwrap_or_else(|| compute_activation_exit_epoch(current_epoch));
+
+        let exit_queue_churn = self
+            .validators
+            .iter()
+            .filter(|v| v.exit_epoch == exit_queue_epoch)
+            .count();
+
+        if exit_queue_churn >= self.get_validator_churn_limit() as usize {
+            exit_queue_epoch += 1;
+        }
+
+        // Set validator exit epoch and withdrawable epoch
+        if let Some(validator) = self.validators.get_mut(index as usize) {
+            validator.exit_epoch = exit_queue_epoch;
+            validator.withdrawable_epoch =
+                validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY;
+        }
+    }
+
+    /// Slash the validator with index ``slashed_index``
+    pub fn slash_validator(
+        &mut self,
+        slashed_index: usize,
+        whistleblower_index: Option<usize>,
```

Comment:
```suggestion
        slashed_index: u64,
        whistleblower_index: Option<u64>,
```

these should be u64's as that is how they are defined in the spec, also then there will be less type conversions in this function so it will be cleaner to read